src/NUnitFramework/framework/Constraints/Constraint.cs

```diff
@@ -54,8 +54,9 @@ protected Constraint(params object[] args)
 
             _displayName = new Lazy<string>(() =>
             {
-                var displayName = this.GetType().Name;
-                if (displayName.EndsWith("`1", StringComparison.Ordinal) || displayName.EndsWith("`2", StringComparison.Ordinal))
+                var type = this.GetType();
+                var displayName = type.Name;
+                if (type.IsGenericType)
```

Comment:
In the other PR, he mentions that he was building using Cake from the command line. On a Mac, IsRunningOnWindows is obviously false which means it will build as if it is Linux which does not include the portable build.